neqo-http3/src/hframe.rs

```diff
@@ -314,39 +317,26 @@ impl HFrameReader {
                 header_block: dec.decode_remainder().to_vec(),
             },
             H3_FRAME_TYPE_CANCEL_PUSH => HFrame::CancelPush {
-                push_id: match dec.decode_varint() {
-                    Some(v) => v,
-                    _ => return Err(Error::NotEnoughData),
-                },
+                push_id: dec.decode_varint().ok_or(Error::HttpFrame)?,
             },
             H3_FRAME_TYPE_SETTINGS => {
                 let mut settings = HSettings::default();
-                settings.decode_frame_contents(&mut dec)?;
+                settings
+                    .decode_frame_contents(&mut dec)
+                    .map_err(|_| Error::HttpFrame)?;
```

Comment:
You can avoid needing the mapping here by implementing From.

Reply:
Ok, I am going to ask a stupid question: can you explain what you mean?

Reply:
`map_err()` is used to translate `Result<T, A>` to `Result<T, B>`.  However, the `?` operator also does the same automatically.  If the function returns `Result<T, B>` and there is an implementation of `Into<B>` on `A` or `From<A>` on `B`, then the conversion is automatic.

But all that doesn't apply here: because the two types are the same.  I think that this is case where you should just change `decode_frame_contents` to return the right type of error.  Then you can drop `map_err` altogether.

Reply:
That was the reason I was asking. I wanted to know if I miss some functionality of From (I do not know what I do not know). 
I will do this in a separate PR. I want to be sure that i am not breaking something.